### ISSUE TITLE
Bug: The token type of null should be `Null` instead of `Keyword`

### ISSUE THREAD

**Issue by JLHwung** [typescript-eslint/typescript-eslint#11027]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Playground Link

https://typescript-eslint.io/play/#ts=5.6.2&showAST=es&fileType=.tsx&code=HYVwNmQ&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=true

### Repro Code

```TypeScript
null
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

`.ast.tokens[0].type` should be `Null`.

### Actual Result

`.ast.tokens[0].type` is `Keyword`.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | 8.28.0 |

**Comment by JoshuaKGoldberg** [comment]
Sorry, I'm still not 100% familiar with these nuances - could you please explain why the token type should be `"null"` instead of `"keyword"`? `null` _is_ a keyword. Other keywords such as `instanceof` have `"keyword"` types too:

```ts
"a" instanceof "b"
```

https://typescript-eslint.io/play/#ts=5.6.2&showAST=es&fileType=.tsx&code=EQQ2AIEsDsGcBcTQMYFMD2AzcwBGwg&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=true

```js
{
  type: "Keyword",
  loc: {end, start},
  range: [4, 14],
  value: "instanceof"
},
```

**Comment by JLHwung** [comment]
> could you please explain why the token type should be "null" instead of "keyword"?

Good question. I don't know the design choice behind it. Either a null type or a keyword type look good to me personally. But this is the espree behaviour ([AST explorer](https://astexplorer.net/#/gist/4282c0e22cfb8b275e1965844afa086e/a231a333dd7e8e5fc08978a5e8b69bc1bba7315b)), here is the espree output for the input source `null`:

```json
{
  "type": "Program",
  "start": 0,
  "end": 4,
  "range": [
    0,
    4
  ],
  "body": [
    {
      "type": "ExpressionStatement",
      "start": 0,
      "end": 4,
      "range": [
        0,
        4
      ],
      "expression": {
        "type": "Literal",
        "start": 0,
        "end": 4,
        "range": [
          0,
          4
        ],
        "value": null,
        "raw": "null"
      }
    }
  ],
  "sourceType": "module",
  "tokens": [
    {
      "type": "Null",
      "value": "null",
      "start": 0,
      "end": 4,
      "range": [
        0,
        4
      ]
    }
  ]
}
```
Since espree is the canonical implementation of estree, I think we should just align to it.
